lib/add_magic_comment.rb

```diff
@@ -16,11 +16,20 @@ module AddMagicComment
 
   def self.process(argv)
     directory = argv.first || Dir.pwd
+    ignore_pattern = Regexp.new(argv[1]) rescue nil
+    # Allow providing as either a Regex or a String
+    if ignore_pattern && ignore_pattern.is_a?(String)
```

Comment:
`argv[1]` will never raise an exception, though, so you can skip the `rescue nil`. `Array#[]` will simply return `nil` if the index is out of bounds.